src/main/java/com/google/devtools/build/remote/client/CacheNotFoundException.java

```diff
@@ -0,0 +1,36 @@
+// Copyright 2018 The Bazel Authors. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.devtools.build.remote.client;
+
+
+import com.google.devtools.remoteexecution.v1test.Digest;
+import java.io.IOException;
+
+/**
+ * An exception to indicate cache misses.
+ * TODO(olaola): have a class of checked RemoteCacheExceptions.
+ */
+public final class CacheNotFoundException extends IOException {
+  private final Digest missingDigest;
+
+  CacheNotFoundException(Digest missingDigest) {
+    super("Missing digest: " + missingDigest);
+    this.missingDigest = missingDigest;
+  }
+
+  public Digest getMissingDigest() {
+    return missingDigest;
+  }
+}
```

Comment:
Add newline. We should find a way to run all this through the Google Java linter.

Reply:
Ooh, found it.

On a corp machine in the root of your git directory, you can run this:
/home/build/nonconf/google3/tools/java/checkstyle/gcheckstyle.sh *

It will check style for conformance to Google guidelines

Reply:
Ah I totally forgot about running a linter.
I think I'll try this: https://github.com/google/google-java-format as well as that

Reply:
Done